cmd/traffic_cache_tool/CacheScan.cc

```diff
@@ -324,7 +324,16 @@ CacheScan::get_alternates(const char *buf, int length)
                         std::string(url->m_ptr_host, url->m_len_host) + ":" + std::string(url->m_ptr_port, url->m_len_port) + "/" +
                         std::string(url->m_ptr_path, url->m_len_path) + ";" + std::string(url->m_ptr_params, url->m_len_params) +
                         "?" + std::string(url->m_ptr_query, url->m_len_query);
-      std::cout << str << std::endl;
+
+      if (search) {
+        std::string str_host =
```

Comment:
OTSOTPR - this would be an excellent example of how `BufferWriter` is cleaner and more efficient.